Obsidian/Hosting/DefaultServerStartup.cs

```diff
@@ -0,0 +1,116 @@
+﻿using System.IO;
+using System.Threading;
+
+namespace Obsidian.Hosting;
+
+/// <summary>
+/// A default <see cref="IServerEnvironment"/> implementation aimed for Console applications.
+/// Loads the server configuration and worlds using the current working directory and
+/// forwards commands from the standard input to the server.
+/// 
+/// Use the <see cref="Create"/> method to create an instance.
+/// </summary>
+public sealed class DefaultServerEnvironment : IServerEnvironment
+{
+    public bool ServerShutdownStopsProgram { get; } = true;
+    public ServerConfiguration Configuration { get; }
+    public List<ServerWorld> ServerWorlds { get; }
+
+    private DefaultServerEnvironment(bool serverShutdownStopsProgram, ServerConfiguration configuration, List<ServerWorld> serverWorlds)
+    {
+        ServerShutdownStopsProgram = serverShutdownStopsProgram;
+        Configuration = configuration;
+        ServerWorlds = serverWorlds;
+    }
+
+    /// <summary>
+    /// Provide server commands using the Console.
+    /// </summary>
+    /// <param name="server"></param>
+    /// <param name="cToken"></param>
+    /// <returns></returns>
+    public async Task ProvideServerCommands(Server server, CancellationToken cToken)
+    {
+        while (!cToken.IsCancellationRequested)
+        {
+            var input = Console.ReadLine();
+            if (input == null) continue;
+            await server.ExecuteCommand(input);
+        }
+    }
+     
+    /// <summary>
+    /// Create a <see cref="DefaultServerEnvironment"/> asynchronously.
+    /// </summary>
+    /// <returns></returns>
+    public static async Task<DefaultServerEnvironment> Create()
+    {
+        var config = await LoadServerConfiguration();
+        var worlds = await LoadServerWorlds();
+        return new DefaultServerEnvironment(true, config, worlds);
+    }
+
+    private static async Task<ServerConfiguration> LoadServerConfiguration()
+    {
+        if (!Directory.Exists("config"))
+            Directory.CreateDirectory("config");
+
+        var configFile = new FileInfo(Path.Combine("config", "main.json"));
+
+        if (configFile.Exists)
+        {
+            using var configFileStream = configFile.OpenRead();
+            var c = await configFileStream.FromJsonAsync<ServerConfiguration>();
+            return c ?? throw new Exception("Server config file exists, but is invalid. Is it corrupt?");
+        }
+
+        var config = new ServerConfiguration();
+
+        using var fs = configFile.Create();
+
+        await config.ToJsonAsync(fs);
+
+        Console.WriteLine($"Created new configuration file for Server");
+        Console.WriteLine($"Please fill in your config with the values you wish to use for your server.\n");
+        Console.ForegroundColor = ConsoleColor.Green;
+        Console.WriteLine(configFile.FullName);
```

Comment:
I wonder if it's possible to update this all to use our hosting logger?

Reply:
Just to be consistent and all.

Reply:
It should be, yes. Let me take a look.

Reply:
On second thought, this is not really Server code. You can just look at this as UI code by the startup application, so it should not be using a logger from Obsidian.